coco_architecture.md

```diff
@@ -0,0 +1,177 @@
+  while it is in use within the TEE
+- **Code integrity:** Unauthorized entities cannot add, remove, or alter code
+  executing in the TEE
+
+# Confidential Computing (CC) Building blocks
+
+## Overview of the key components of a confidential computing solution
+
+The following diagrams shows how the different building blocks for CC come together:
+
+![CC_all_blocks](https://github.com/confidential-containers/documentation/blob/main/images/CC_all_blocks.jpg)
+
+Let’s start by describing the lower level blocks of the CC solution (**colored blue**):
+
+- **Infra layer** - be it on-premises (bare metal, VMware etc.) or public clouds
+  (AWS, Azure, GCP, etc.)
+  - **Hardware with CC support** - this includes 2 distinct models:
+    - **VM-based TEEs**  - In this model, memory is encrypted along a
+      traditional VM boundary running on top of a VMM. AMD SEV-SNP, Intel TDX,
+      IBM Secure Execution and Protected Execution Functionality (PEF) are
+      examples of VM-based TEEs.
+    - **Process-based TEEs** - In this model, a process that needs to run
+      securely is divided into two components: trusted and untrusted.
+      The trusted component resides in encrypted memory and handles confidential
+      computing, while the untrusted component interfaces with the operating
+      system and propagates I/O from encrypted memory to the rest of the system.
+      Intel SGX is an example of a process-based TEE.
+- **Hypervisor** - this includes hypervisors such as QEMU/KVM, cloud hypervisor,
+  public cloud provider hypervisors etc...
+
+The **Confidential Computing services block** contains a number of services
+(**colored pink**) which are required for creating a holistic CC platform which
+the customer can then use to build their solution. This currently includes the
+following services:
+
+- **Attestation service** -  The primary purpose of the attestation service is
+  to validate the evidence provided by the hardware TEE. This is the *Verifier*,
+  as defined in the [RATS architecture](https://datatracker.ietf.org/doc/html/draft-ietf-rats-architecture-21).
+- **Key Broker Service (KBS)** -  The KBS is the *Relying Party*, as defined by
+  the [RATS architecture](https://datatracker.ietf.org/doc/html/draft-ietf-rats-architecture-21).
+  Following are its primary functions:
+  - Receive evidence from the *Attester* (confidential VM or container) via a
+    challenge-response protocol.
+  - Relay the evidence to the Attestation Service for verification.
+  - Apply appraisal policy for the returned Attestation Results to assess the
+    trustworthiness of the *Attester*.
+  - Interact with the Key Management Service to retrieve the keys and then
+    send them back to the *Attester*.
+- **Key management service** - A service for securely storing, managing and
+  backing up of cryptographic keys used by applications and users.
+- **Image build service** - A service that is used to build confidential
+  container or VM images for end users.
```

Comment:
I worry that people may assume a single build service that builds both the confidential containers and the VM Images and that a single end user persona or role drives this build service?
 Not sure if we can change this a little? 
Perhaps "Services used to build confidential containers or VM images"